File: spring/src/main/java/com/playground/model/signalComment.java
package com.playground.model;

import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;

@Entity
@Table(name = "signal_comment")
@EntityListeners(AuditingEntityListener.class)
public class signalComment {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    @OneToOne
    private Comment comment;
    @OneToOne
    private User author;
    private String description;

    public signalComment() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Comment getComment() {
        return comment;
    }

    public void setComment(Comment comment) {
        this.comment = comment;
    }

    public User getAuthor() {
        return author;
    }

    public void setAuthor(User author) {
        this.author = author;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: mobile/lib/entities/SignalComment.dart

import 'package:Playground/entities/Comment.dart';
import 'package:Playground/entities/User.dart';

class SignalComment {

  int id;
  Comment comment;
  User author;
  String description;

  SignalComment({this.id, this.comment, this.author, this.description});

  factory SignalComment.getDefault() {
    return new SignalComment(
      id: 0,
      description: "",
      comment: null,
      author: null
    );
  }

  Map<String, dynamic> toJson() {
    return <String, dynamic> {
      'id' : this.id,
      'description' : this.description,
      'author' : this.author.toJson(),
      'comment' : this.comment.toJson()
    };
  }

}